src/BaseListView.php

```diff
@@ -208,56 +242,83 @@ public function renderEmpty()
     /**
      * Renders the summary text.
      */
-    public function renderSummary()
+    public function renderSummary(): string
     {
-        $count = $this->dataProvider->getCount();
+        $count = $this->dataReader->count();
         if ($count <= 0) {
             return '';
         }
+        // TODO fix that
+        $language = 'language';
+
         $summaryOptions = $this->summaryOptions;
         $tag = ArrayHelper::remove($summaryOptions, 'tag', 'div');
-        if (($pagination = $this->dataProvider->getPagination()) !== false) {
-            $totalCount = $this->dataProvider->getTotalCount();
-            $begin = $pagination->getPage() * $pagination->pageSize + 1;
+        if (($pagination = $this->paginator) !== null) {
+            $totalCount = $this->dataReader->count();
+            $begin = 0; //$pagination->getCurrentPageSize() * $pagination->pageSize + 1;
             $end = $begin + $count - 1;
             if ($begin > $end) {
                 $begin = $end;
             }
-            $page = $pagination->getPage() + 1;
-            $pageCount = $pagination->pageCount;
+            $page = 0; //$pagination->getPage() + 1;
```

Comment:
?

Reply:
`getPage` doen't exist in PaginatorInterface.

Reply:
Correct. There are two pagination types now:

1. Yii 2 like offset pagination. You can get current page.
2. No-offset pagination. You can not get current page.

Data widgets should support both i.e. there should be separate pagination widgets for these types.

Reply:
The second paginator may be added when:
1. The first paginator will work fine
2. The `keyset` paginator will be used by some package

Note: depend on implementation instead of interface is a sing of bad smell.
I think that the `PaginatorInterface` was poorly designed. 
`Paginator` (as a derivative of `page`) shoud contain `*Page()` methods.
`Keyset` (as a stream) shouldn't contain `*Page()` methods.

Anyway `KeysetPaginator` will be supported when `Dataview` will work fine with `OffsetPaginator`